### DISCUSSION TITLE
[HELP] How to "carve" mesh and fill holes after cutting?

### DISCUSSION THREAD

**Discussion by byaenh** [marcomusy/vedo#919] [Q&A] [answered]
Hi everyone, 

My team and I are trying to use Vedo to carve a 3D mesh of a tooth in a certain shape. However, the issue that we are running into is that once we cut our mesh (e.g. using cut_with_plane), it's very difficult to accurately fill the gaps that we are cutting away. At the moment, we are using mesh.slice() to fill our holes. However, that only works if we are cutting with a plane and furthermore, it doesn't work where the boundary for the slice is not connected. 

We were wondering if anyone knew of good Vedo functions that could help us to carve the tooth and re-fill the gaps? Ideally, if there was a Vedo function that could make our 3D mesh outline into a solid shape or if there's a good filling holes function which can link gaps between points with a flat surface, that would be amazing.

**Comment by marcomusy** [comment, marked as answer]
Hi @byaenh 
sorry for the late reply, the `mesh.slice()`  should actually work for disconnected boundaries:

```py
from vedo import *
msh = Mesh(dataurl+"bunny.obj").alpha(0.1).wireframe()
mslice = msh.slice(normal=[0,1,0.3], origin=[0,0.16,0])
mslice.c('purple5')
show(msh, mslice, axes=1)
```

 
![Screenshot from 2023-09-07 17-11-44](https://github.com/marcomusy/vedo/assets/32848391/72ebe9d8-8180-4973-a941-1368fd961ce7)

Another option is to use `generate_mesh` for a 2D-projectable planar contour, see example
`vedo/examples/advanced/capping_mesh.py`
